### PR TITLE
Add pre auth check step

### PR DESCRIPTION
**Pull request by yathindrakodithuwakku**
### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/home.jsp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<!--
+* Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/console/src/home.jsp:2]:
```suggestion
* Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/console/src/home.jsp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+-->
+
+<%= htmlWebpackPlugin.options.importUtil %>
+<%= htmlWebpackPlugin.options.importTenantPrefix %>
+<%= htmlWebpackPlugin.options.importSuperTenantConstant %>
+
+<script>
```
Comment (brionmario) [inline apps/console/src/home.jsp:23]: Can't we put this inside the body?

File: apps/console/src/index.jsp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <!--
-* Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+* Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/console/src/index.jsp:2]:
```suggestion
* Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/console/src/index.jsp [hunk 4]
```diff
@@ -16,36 +16,72 @@
...
+                overrideWellEndpointConfig: true
+            }
+
+            auth.initialize(authConfig);
+
+            auth.trySignInSilently().then(res => {
+                if(res === false) {
+                    auth.signIn();
+                } 
+                else {
```
Comment (brionmario) [inline apps/console/src/index.jsp:78]:
```suggestion
                if () {
                } else {
```

File: apps/console/src/index.jsp [hunk 5]
```diff
@@ -16,36 +16,72 @@
...
+                    wellKnownEndpointURL: undefined,
+                },
+                enablePKCE: true,
+                overrideWellEndpointConfig: true
+            }
+
+            auth.initialize(authConfig);
+
+            auth.trySignInSilently().then(res => {
+                if(res === false) {
```
Comment (brionmario) [inline apps/console/src/index.jsp:75]: Is there a possibility of res being any other falsy value than `false`?
```suggestion
                if(res === false) {
```

File: apps/myaccount/src/home.jsp [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<!--
+* Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/myaccount/src/home.jsp:2]:
```suggestion
* Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/console/src/index.jsp (hunk 5 above)
Comment (yathindrakodithuwakku) [reply apps/console/src/index.jsp:75]: No the response can be boolean or user info object
References: https://github.com/asgardeo/asgardeo-auth-spa-sdk/blob/5d60a661508ab749aa0a6ad08a7505656f2f7605/lib/src/models/client.ts#L65

File: apps/console/src/home.jsp (hunk 2 above)
Comment (yathindrakodithuwakku) [reply apps/console/src/home.jsp:23]: Kept here as this is an import statement

**Review by brionmario — APPROVED** [review]

**Pull request merged**